src/Arcus.WebApi.Security/Authentication/CertificateAuthenticationFilter.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using GuardNet;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Arcus.WebApi.Security.Authentication
+{
+    /// <summary>
+    /// Authentication filter to secure HTTP requests by allowing only certain values in the client certificate.
+    /// </summary>
+    public class CertificateAuthenticationFilter : IAuthorizationFilter
```

Comment:
We should pull in `ILogger` in here and provide information traces of what is going on. This should help customers analyse issues.

Scenarios I'm thinking of where I'd like to have information:
1. No client cert was specified
2. X509ValidationRequirement.SubjectName failed and why (incl expected, actual)
2. X509ValidationRequirement.IssuerName failed and why (incl expected, actual)
2. X509ValidationRequirement.Thumbprint failed and why (incl expected, actual)

Reply:
Yes, will do; is this something that from the services should be retrieved?

Reply:
Yes, will do!